### PR TITLE
[release-4.5] Add component to render metrics inside ROKS clusters

### PR DESCRIPTION
**Pull request by csrwng**
Backport of #58

### CODE REVIEW

File: assets/roks-metrics/roks-metrics-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        openshift.io/restartedAt: "{{ .RestartDate }}"
+{{ end }}
+    spec:
+      tolerations:
+        - key: "multi-az-worker"
+          operator: "Equal"
+          value: "true"
+          effect: NoSchedule
+      affinity:
+        podAntiAffinity:
```
Comment (jonesbr17) [inline assets/roks-metrics/roks-metrics-deployment.yaml:31]: This is intentionally left blank?

File: cluster.yaml.example [hunk 2]
```diff
@@ -148,3 +148,4 @@ apiserverLivenessPath: livez?exclude=etcd
 controlPlaneOperatorImage: registry.svc.ci.openshift.org/ibm-roks/ibm-roks-4.4:control-plane-operator
 controlPlaneOperatorSecurity: 1001
 masterPriorityClass: ''
+roksMetricsImage: registry.svc.ci.openshift.org/hypershift-toolkit/ibm-roks-4.6:roks-metrics
```
Comment (jonesbr17) [inline cluster.yaml.example:151]: I assume we're going to have to coordinate some work on our side to ensure we provide this image. I don't see any logic to skip the metrics setup if this isn't provided.

File: pkg/render/manifests.go [hunk 3]
```diff
@@ -62,6 +62,7 @@ func (c *clusterManifestContext) setupManifests(externalOauth, includeRegistry b
 	if includeRegistry {
 		c.registry()
 	}
+	c.roksMetrics()
```
Comment (jonesbr17) [inline pkg/render/manifests.go:65]: To go along with my comment above, we'll either need an `if ROKSMetricsImage != ""` here, or hold off on merging this until our code is prepared to pass the image over.

**Review by jonesbr17 — CHANGES_REQUESTED** [review]

File: pkg/render/manifests.go (hunk 3 above)
Comment (csrwng) [reply pkg/render/manifests.go:65]: hmm, you're right, I think we should make this optional until the image is specified. Will change.

File: assets/roks-metrics/roks-metrics-deployment.yaml (hunk 1 above)
Comment (csrwng) [reply assets/roks-metrics/roks-metrics-deployment.yaml:31]: we don't need the affinity part, will remove

**Comment by csrwng** [comment]
@jonesbr17 made requested changes, thank you for the review!

**Review by jonesbr17 — APPROVED** [review]

**Pull request merged**